docs/user-guide/optimization.md

```diff
@@ -0,0 +1,109 @@
+# Optimization
+
+Cubed will automatically optimize the computation graph before running it. This can reduce the number of tasks in the plan, and the amount of intermediate IO, both of which speed up the computation.
```

Comment:
From this intro it is unclear why I would ever not want to fuse tasks. Are there situations in which fusing would sacrifice memory safety? Or is it merely a trade-off between asking single tasks to read more data and potentially having more IO steps in the pipeline?

Reply:
> Are there situations in which fusing would sacrifice memory safety?

No, fusing should always respect memory limits.

> Or is it merely a trade-off between asking single tasks to read more data and potentially having more IO steps in the pipeline?

Yes that's it.